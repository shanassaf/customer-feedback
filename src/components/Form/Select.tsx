import React from 'react';
import styled from 'styled-components';

interface Props {
	setFeedback: (feedback: Feedback) => void;
	feedback: Feedback;
}

const StyledSelect = styled.select`
	padding: 10px;
	width: 50px;
	cursor: pointer;

	:hover {
		border-color: #1b047c;
	}
`;
const StyledOption = styled.option``;

export const Select = ({ setFeedback, feedback }: Props): JSX.Element => {
	return (
		<StyledSelect
			required
			name="rating"
			value={feedback.rating}
			onChange={(ev) => {
				setFeedback({ ...feedback, rating: ev.target.value });
			}}
		>
			<StyledOption value="0">0</StyledOption>
			<StyledOption value="1">1</StyledOption>
			<StyledOption value="2">2</StyledOption>
			<StyledOption value="3">3</StyledOption>
			<StyledOption value="4">4</StyledOption>
			<StyledOption value="5">5</StyledOption>
		</StyledSelect>
	);
};
