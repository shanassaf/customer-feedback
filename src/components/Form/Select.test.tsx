import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Select } from './Select';

describe('Select', () => {
	const feedback = {
		name: '',
		email: '',
		comment: '',
		rating: '0',
	} as Feedback;
	const props = {
		setFeedback: jest.fn(),
		feedback,
	};
	test('component renders correctly', () => {
		const { asFragment } = render(<Select {...props} />);
		expect(asFragment()).toMatchSnapshot();
	});
	test('component calls setComment on change', () => {
		const { getByRole } = render(<Select {...props} />);
		const select = getByRole('combobox');
		fireEvent.change(select, { target: { value: '3' } });
		expect(props.setFeedback).toHaveBeenCalledWith({ ...feedback, rating: '3' });
	});
});
