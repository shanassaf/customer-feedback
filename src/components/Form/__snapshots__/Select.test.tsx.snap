// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component renders correctly 1`] = `
<DocumentFragment>
  <select
    class="sc-bdnylx kYqwRh"
    name="rating"
    required=""
  >
    <option
      class="sc-gtssRu"
      value="0"
    >
      0
    </option>
    <option
      class="sc-gtssRu"
      value="1"
    >
      1
    </option>
    <option
      class="sc-gtssRu"
      value="2"
    >
      2
    </option>
    <option
      class="sc-gtssRu"
      value="3"
    >
      3
    </option>
    <option
      class="sc-gtssRu"
      value="4"
    >
      4
    </option>
    <option
      class="sc-gtssRu"
      value="5"
    >
      5
    </option>
  </select>
</DocumentFragment>
`;
